src/z_dateformat.erl

```diff
@@ -304,14 +305,21 @@ tag_to_value($w, Date, _, _Options) ->
 tag_to_value($W, {Y,M,D}, _, _Options) ->
    integer_to_list(year_weeknum(Y,M,D));
 
+% Year with at least 4 digits, padded with zeroes (e.g. '0003')
+tag_to_value($x, {Y, M, D}, T, Options) when Y < 0 ->
+    "-" ++ tag_to_value($x, {abs(Y), M, D}, T, Options);
+tag_to_value($x, {Y, _, _}, _, _Options) ->
+    Length = integer_to_list(max(4, length(integer_to_list(Y)))),
+    lists:flatten(io_lib:format("~" ++ Length ++ "..0B", [Y]));
```

Comment:
Is there a more elegant way to do this?

Reply:
Or just a guard on the `Y` and a special case where your just show the number as-is if longer than 4 digits?

```erlang
tag_to_value($x, {Y, _, _}, _, _Options) when Y >= 1000; Y =< -1000 ->
```

Reply:
Right, as I proposed above. Done!